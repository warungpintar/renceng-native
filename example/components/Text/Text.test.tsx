import React from 'react';
import {render} from '@testing-library/react-native';
import {Text, colors} from '@warungpintar/renceng-native';

describe('Text', () => {
  test('basic', () => {
    const props = {
      testID: 'my-text',
      fontSize: 12,
    };
    const {queryByTestId, toJSON} = render(
      <Text {...props}>Text Content</Text>,
    );
    const textQuery = queryByTestId('my-text');
    const style = textQuery.props.style[0];
    const content = textQuery.children.toString();
    expect(style.color).toMatch('#161616');
    expect(style.fontSize).toEqual(12);
    expect(content.match('Text Content'));
    expect(toJSON()).toMatchSnapshot();
  });
  test('category', () => {
    const props = {
      testID: 'my-text',
      category: 'md',
    };
    const {queryByTestId, toJSON} = render(
      <Text {...props}>Text Content</Text>,
    );
    const textQuery = queryByTestId('my-text');
    const style = textQuery.props.style[0];
    const content = textQuery.children.toString();
    expect(style.color).toMatch('#161616');
    expect(style.fontSize).toEqual(14);
    expect(content.match('Text Content'));
    expect(toJSON()).toMatchSnapshot();
  });
  test('align', () => {
    const props = {
      testID: 'my-text',
    };
    const {queryByTestId, toJSON} = render(
      <Text {...props} textAlign="left">
        Text Content
      </Text>,
    );
    const textQuery = queryByTestId('my-text');
    const style = textQuery.props.style[0];
    const content = textQuery.children.toString();
    expect(style.color).toMatch('#161616');
    expect(style.textAlign).toMatch('left');
    expect(content.match('Text Content'));
    expect(toJSON()).toMatchSnapshot();
  });
  test('color', () => {
    const props = {
      testID: 'my-text',
      color: colors.primary,
    };
    const {queryByTestId, toJSON} = render(
      <Text {...props}>Text Content</Text>,
    );
    const textQuery = queryByTestId('my-text');
    const style = textQuery.props.style[0];
    const content = textQuery.children.toString();
    expect(style.color).toMatch('#FECF28');
    expect(content.match('Text Content'));
    expect(toJSON()).toMatchSnapshot();
  });
});
