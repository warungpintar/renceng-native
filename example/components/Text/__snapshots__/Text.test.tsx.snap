// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text align 1`] = `
<Text
  color="black"
  fontFamily="OpenSans"
  style={
    Array [
      Object {
        "color": "black",
        "fontFamily": "OpenSans",
        "textAlign": "left",
      },
      undefined,
    ]
  }
  testID="my-text"
  textAlign="left"
>
  Text Content
</Text>
`;

exports[`Text basic 1`] = `
<Text
  color="black"
  fontFamily="OpenSans"
  fontSize={12}
  style={
    Array [
      Object {
        "color": "black",
        "fontFamily": "OpenSans",
        "fontSize": 12,
      },
      undefined,
    ]
  }
  testID="my-text"
>
  Text Content
</Text>
`;

exports[`Text category 1`] = `
<Text
  category="md"
  color="black"
  fontFamily="OpenSans"
  style={
    Array [
      Object {
        "color": "black",
        "fontFamily": "OpenSans",
        "fontSize": 14,
      },
      undefined,
    ]
  }
  testID="my-text"
>
  Text Content
</Text>
`;

exports[`Text color 1`] = `
<Text
  color="#FECF28"
  fontFamily="OpenSans"
  style={
    Array [
      Object {
        "color": "#FECF28",
        "fontFamily": "OpenSans",
      },
      undefined,
    ]
  }
  testID="my-text"
>
  Text Content
</Text>
`;
